int32[] steps
uint64 duration_us